packages/ckeditor5-image/src/image/imageediting.js

```diff
@@ -125,12 +148,46 @@ export default class ImageEditing extends Plugin {
 //
 // @private
 // @param {module:engine/view/downcastwriter~DowncastWriter} writer
+// @param {'block'|'inline'} imageType The type of created image.
 // @returns {module:engine/view/containerelement~ContainerElement}
-export function createImageViewElement( writer ) {
+export function createImageViewElement( writer, imageType ) {
 	const emptyElement = writer.createEmptyElement( 'img' );
-	const figure = writer.createContainerElement( 'figure', { class: 'image' } );
 
-	writer.insert( writer.createPositionAt( figure, 0 ), emptyElement );
+	const container = imageType === 'block' ?
+		writer.createContainerElement( 'figure', { class: 'image' } ) :
+		writer.createContainerElement( 'span', { class: 'image-inline' } );
+
+	writer.insert( writer.createPositionAt( container, 0 ), emptyElement );
 
-	return figure;
+	return container;
+}
+
+// {@link module:engine/view/matcher~Matcher} pattern. Returns function which checks if a given element is `<img>` element that is placed
+// inside the element of a provided type.
```

Comment:
I think this description could be simpler:

```suggestion
// A function returning a {@link module:engine/view/matcher~Matcher} callback for a particular type of View images. 
```